### PR TITLE
fixed Prevent validating tx signature again #1435

### PR DESCRIPTION
**Pull request by xuecanlong**
1. add ValidatorsCacheTime on config
2. optimization envelope validation and reject action validation

### CODE REVIEW

File: actpool/actpool.go [hunk 1]
```diff
@@ -8,11 +8,10 @@ package actpool
 
 import (
 	"context"
-	"sync"
-
 	"github.com/iotexproject/iotex-core/pkg/prometheustimer"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
+	"sync"
```
Comment (zjshen14) [inline actpool/actpool.go]: wrong import group

File: action/protocol/actpool.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+)
+
+// ActPoolInterface is the interface for actpool
+type ActPoolInterface struct {
+	mu      sync.RWMutex
```
Comment (zjshen14) [inline action/protocol/actpool.go]: What's the use of `mu`?

File: action/protocol/protocol.go [hunk 3]
```diff
@@ -121,3 +121,9 @@ func (m *MockChainManager) ProductivityByEpoch(epochNum uint64) (uint64, map[str
 func (m *MockChainManager) ExecuteContractRead(caller address.Address, ex *action.Execution) ([]byte, *action.Receipt, error) {
 	return nil, nil, nil
 }
+
+// ActPoolManager defines the actpool interface
+type ActPoolManager interface {
```
Comment (zjshen14) [inline action/protocol/protocol.go]: Shall we define this in the blockchain package?

File: action/protocol/actpool.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	"sync"
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+)
+
+// ActPoolInterface is the interface for actpool
+type ActPoolInterface struct {
```
Comment (zjshen14) [inline action/protocol/actpool.go]: Why is `ActPoolInterface` necessary? Can we just use ActPoolManager interface instead?

**Comment by xuecanlong** [comment]
Like this? I am confused.

**Pull request merged**